src/pool-cl/CLPositionManager.sol

```diff
@@ -300,54 +236,48 @@ contract CLPositionManager is
         }
         _mint(owner, tokenId);
 
-        // fee delta can be ignored as this is a new position
-        (BalanceDelta liquidityDelta,) = _modifyLiquidity(config, liquidity.toInt256(), bytes32(tokenId), hookData);
-        liquidityDelta.validateMaxIn(amount0Max, amount1Max);
-        positionConfigs[tokenId].setConfigId(config.toId());
+        // Initialize the position info
+        CLPositionInfo info = CLPositionInfoLibrary.initialize(poolKey, tickLower, tickUpper);
+        positionInfo[tokenId] = info;
 
-        // cache the poolKey in _poolIdToPoolKey
-        PoolId poolId = config.poolKey.toId();
-        if (_poolIdToPoolKey[poolId].parameters == bytes32(0)) {
-            _poolIdToPoolKey[poolId] = config.poolKey;
+        // Store the poolKey if it is not already stored.
+        // if poolManager is address(0), it means the pool is not initialized yet
+        bytes25 poolId = info.poolId();
+        if (address(poolKeys[poolId].poolManager) == address(0)) {
+            poolKeys[poolId] = poolKey;
         }
```

Comment:
this seems to be cheaper in gas 

```
if (poolKeys[poolId].parameters == bytes32(0)) {
  poolKeys[poolId] = poolKey;
}
```

Reply:
verified on my side as well 😂  interesting